File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-5.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        res = root.val
        count = k
        def _dfs(root):
            nonlocal count, res
            if not root:
                return
            _dfs(root.left)
            count -= 1
            if count == 0:
                res = root.val
                return
            _dfs(root.right)
        _dfs(root)
        return res

File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-5.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func inorderTraversal(root *TreeNode) []int {
    if root == nil{
        return []int{}
    }
    var res []int
    dfs(root, &res)
    return res
}

func dfs(root *TreeNode, res *[]int){
    if root == nil{
        return
    }
    dfs(root.Left, res)
    *res = append(*res, root.Val)
    dfs(root.Right, res)
}
